modules/data.remote/R/MODIS_LC_prep.R

```diff
@@ -0,0 +1,178 @@
+#' Prepare MODIS land cover data for the SDA workflow.
+#'
+#' @param site_info Bety list of site info including site_id, lon, and lat.
+#' @param time_points A vector contains each time point within the start and end date.
+#' @param outdir Where the final CSV file will be stored.
+#' @param export_csv Decide if we want to export the CSV file.
+#' @param qc.filter decide if we want to filter data by the qc band.
+#'
+#' @return A data frame containing MODIS land cover types for each site and each time step.
+#' @export
+#' 
+#' @examples
+#' @author Dongchen Zhang
+#' @importFrom magrittr %>%
+MODIS_LC_prep <- function(site_info, time_points, outdir = NULL, export_csv = FALSE, qc.filter = FALSE){
+  #initialize future parallel computation.
+  if (future::supportsMulticore()) {
+    future::plan(future::multicore, workers = 10)
+  } else {
+    future::plan(future::multisession, workers = 10) #10 is the maximum number of requests permitted for the MODIS server.
+  }
+  #if we export CSV but didn't provide any path
+  if(as.logical(export_csv) && is.null(outdir)){
+    PEcAn.logger::logger.info("If you want to export CSV file, please ensure input the outdir!")
+    return(0)
+  }
+  #convert from dates into years.
+  years <- lubridate::year(time_points)
+  #grab previous data to see which site has incomplete observations, if so, download the site for the whole time period.
+  #if we have previous downloaded CSV file
+  if (!is.null(outdir)) {
+    if(file.exists(file.path(outdir, "LC.csv"))){
+      PEcAn.logger::logger.info("Extracting previous MODIS Land Cover file!")
+      Previous_CSV <- utils::read.csv(file.path(outdir, "LC.csv"))
```

Comment:
This sort of cache behavior is unintuitive and isn't documented. Please explain this in the Roxygen

Reply:
fixed.

Reply:
Could you expand a bit more on the fact that the code loads up previous outputs